src/index.ts

```diff
@@ -93,36 +112,169 @@ export class ConsentControl {
      * @param {ITextResources} textResources the text want to be displayed
      */
     public setTextResources(textResources: ITextResources): void {
-        if (textResources.bannerMessageHtml) {
-            this.textResources.bannerMessageHtml = textResources.bannerMessageHtml;
+        for (let key of Object.keys(this.textResources)) {
+            let typeKey = <keyof ITextResources> key;
+
+            if (textResources[typeKey]) {
+                this.textResources[typeKey] = textResources[typeKey];
+            }
+        }
+    }
+
+    /**
+     * Use the passed theme to set the theme property
+     * 
+     * @param {string} name the theme property that we want to set
+     * @param {ITheme} theme the passed theme that we want to display
+     */
+    public createTheme(name: string, theme: ITheme): void {
+        let typeName = <keyof IThemes> name;
+        let currentTheme = <ITheme> this.themes[typeName];
```

Comment:
This behavior will not create a new theme. Change this so it works if I pass a new theme e.g. "ultra-dark"